src/Security/Authentication/Cookies/src/CookieAuthenticationHandler.cs

```diff
@@ -140,12 +140,12 @@ private async Task<AuthenticateResult> ReadCookieTicket()
                 {
                     return AuthenticateResult.Fail("SessionId missing");
                 }
-                _sessionKey = claim.Value;
                 ticket = await Options.SessionStore.RetrieveAsync(_sessionKey);
```

Comment:
```suggestion
                // Only store _sessionKey if it matches an existing session. Otherwise we'll create a new one.
                ticket = await Options.SessionStore.RetrieveAsync(claim.Value);
```